subsys/usb/host/usbh_device.c

```diff
@@ -464,11 +466,19 @@ int usbh_device_init(struct usb_device *const udev)
 		return err;
 	}
 
-	/* FIXME: The port to which the device is connected should be reset. */
-	err = uhc_bus_reset(uhs_ctx->dev);
-	if (err) {
-		LOG_ERR("Failed to signal bus reset");
-		return err;
+	k_mutex_lock(&uhs_ctx->mutex, K_FOREVER);
+	SYS_DLIST_FOR_EACH_NODE(&uhs_ctx->udevs, node) {
+		device_count++;
+	}
```

Comment:
I think you can use `device_count = sys_dlist_len(&uhs_ctx->udevs);` in this case.

[doc](https://docs.zephyrproject.org/latest/doxygen/html/group__doubly-linked-list__apis.html#ga6797f138ad66cfb5ee7918bb12869eac)